tools/mtmd/clip.cpp

```diff
@@ -4059,14 +4075,20 @@ int clip_n_mmproj_embd(const struct clip_ctx * ctx) {
         case PROJECTOR_TYPE_MLP_NORM:
             return ctx->model.mm_3_b->ne[0];
         case PROJECTOR_TYPE_MINICPMV:
-            if (hparams.minicpmv_version == 2) {
-                return 4096;
-            } else if (hparams.minicpmv_version == 3) {
-                return 3584;
-            } else if (hparams.minicpmv_version == 4) {
-                return 3584;
+            // Use actual config value if available, otherwise fall back to hardcoded values
+            if (hparams.minicpmv_projection_dim > 0) {
+                return hparams.minicpmv_projection_dim;
```

Comment:
maybe this doesn't need to be a metadata, you can read it from tensor shape, just like other models